chainstate/src/detail/chainstateref/tx_verifier_storage.rs

```diff
@@ -181,15 +182,31 @@ impl<'a, S: BlockchainStorageWrite, O: OrphanBlocks, V: TransactionVerificationS
 
     fn set_undo_data(
         &mut self,
-        id: Id<Block>,
+        tx_source: TransactionSource,
```

Comment:
I think this is unnecessary. Instead of making it possible to flush mempool tx then panic, why don't we just prevent it through types?

Reply:
I panic only when flushing to ChainstateRef, but at the same time it's ok to flush undo for mempool from one verifier to another